File: server_python/engine/summary_engine.py
level_map_te = {
    'CRITICAL': 'అత్యంత ప్రమాదకరం',
    'HIGH': 'ప్రమాదకరం',
    'MEDIUM': 'మధ్యస్థం',
    'LOW': 'తక్కువ'
}

disease_map_te = {
    'Blast': 'అగ్గి తెగులు',
    'Blight': 'ఎండు తెగులు',
    'Tungro': 'టుంగ్రో తెగులు',
    'Brown Spot': 'గోధుమ రంగు మచ్చ తెగులు'
}

class SummaryEngine:
    def _en_short(self, disease: str, level: str) -> str:
        return f"🌾 PaddyPulse Alert: {level} risk of {disease} detected. Immediate action recommended."

    def _en_long(self, data: dict) -> str:
        reasons = ", ".join(data.get('reasons', []))
        return (f"🌾 *PaddyPulse Health Advisory* 🌾\n\n"
                f"📍 *Field Status:* {data.get('risk_level')} Risk ({data.get('risk_score')}/100)\n"
                f"🦠 *Disease:* {data.get('disease')}\n"
                f"🗺️ *Affected Zones:* {data.get('zones_count')} zones showing pressure\n\n"
                f"✅ *Action Plan:* Spray {data.get('treatment')} ({data.get('dosage')})\n"
                f"⏰ *Spraying Window:* {data.get('spray_window')}\n\n"
                f"💡 *Why:* {reasons}\n\n"
                f"⚠️ Avoid spraying if rain is expected in 2 hours.")

    def _te_short(self, disease: str, level: str) -> str:
        return f"🌾 వరి హెచ్చరిక: మీ పొలంలో {disease} ఆశించే అవకాశం {level} గా ఉంది. వెంటనే తగిన చర్యలు తీసుకోగలరు."

    def _te_long(self, data: dict) -> str:
        reasons = ", ".join(data.get('reasons_te', []))
        return (f"🌾 *వరి ఆరోగ్య సలహా* 🌾\n\n"
                f"📍 *పొలం స్థితి:* {data.get('risk_level_te')} ప్రమాదం ({data.get('risk_score')}/100)\n"
                f"🦠 *తెగులు:* {data.get('disease_te')}\n"
                f"🗺️ *ప్రభావిత ప్రాంతాలు:* {data.get('zones_count')} జోన్లు\n\n"
                f"✅ *చేయవలసిన పని:* {data.get('treatment')} ({data.get('dosage')}) పిచికారీ చేయండి\n"
                f"⏰ *మంచి సమయం:* {data.get('spray_window')}\n\n"
                f"💡 *కారణాలు:* {reasons}\n\n"
                f"⚠️ రాబోయే 2 గంటల్లో వర్షం కురిసే అవకాశం ఉంటే పిచికారీ చేయకండి.")

    def generate(self, data: dict) -> dict:
        risk_level = data.get('risk_level', 'LOW')
        disease = data.get('disease', 'Unknown')
        
        en_short = self._en_short(disease, risk_level)
        en_long = self._en_long(data)

        # Prepare te data
        te_data = data.copy()
        te_data['risk_level_te'] = level_map_te.get(risk_level, risk_level)
        te_data['disease_te'] = disease_map_te.get(disease, disease)
        te_data['reasons_te'] = data.get('reasons_te') or data.get('reasons', [])

        te_short = self._te_short(te_data['disease_te'], te_data['risk_level_te'])
        te_long = self._te_long(te_data)

        return {
            "english_text_short": en_short,
            "english_text_long": en_long,
            "telugu_text_short": te_short,
            "telugu_text_long": te_long
        }

summary_engine = SummaryEngine()

File: server/engine/summary_engine.js
/**
 * Farmer Summary Generator Engine
 * Produces multilingual summaries for WhatsApp and detailed advisory.
 */

const translations = {
    en: {
        short_template: (disease, level) => `🌾 PaddyPulse Alert: ${level} risk of ${disease} detected. Immediate action recommended.`,
        long_template: (data) => `🌾 *PaddyPulse Health Advisory* 🌾\n\n` +
            `📍 *Field Status:* ${data.risk_level} Risk (${data.risk_score}/100)\n` +
            `🦠 *Disease:* ${data.disease}\n` +
            `🗺️ *Affected Zones:* ${data.zones_count} zones showing pressure\n\n` +
            `✅ *Action Plan:* Spray ${data.treatment} (${data.dosage})\n` +
            `⏰ *Spraying Window:* ${data.spray_window}\n\n` +
            `💡 *Why:* ${data.reasons.join(', ')}\n\n` +
            `⚠️ Avoid spraying if rain is expected in 2 hours.`
    },
    te: {
        short_template: (disease, level) => `🌾 వరి హెచ్చరిక: మీ పొలంలో ${disease} ఆశించే అవకాశం ${level} గా ఉంది. వెంటనే తగిన చర్యలు తీసుకోగలరు.`,
        long_template: (data) => `🌾 *వరి ఆరోగ్య సలహా* 🌾\n\n` +
            `📍 *పొలం స్థితి:* ${data.risk_level} ప్రమాదం (${data.risk_score}/100)\n` +
            `🦠 *తెగులు:* ${data.disease}\n` +
            `🗺️ *ప్రభావిత ప్రాంతాలు:* ${data.zones_count} జోన్లు\n\n` +
            `✅ *చేయవలసిన పని:* ${data.treatment} (${data.dosage}) పిచికారీ చేయండి\n` +
            `⏰ *మంచి సమయం:* ${data.spray_window}\n\n` +
            `💡 *కారణాలు:* ${data.reasons_te.join(', ')}\n\n` +
            `⚠️ రాబోయే 2 గంటల్లో వర్షం కురిసే అవకాశం ఉంటే పిచికారీ చేయకండి.`
    }
};

const levelMapTe = {
    'CRITICAL': 'అత్యంత ప్రమాదకరం',
    'HIGH': 'ప్రమాదకరం',
    'MEDIUM': 'మధ్యస్థం',
    'LOW': 'తక్కువ'
};

const diseaseMapTe = {
    'Blast': 'అగ్గి తెగులు',
    'Blight': 'ఎండు తెగులు',
    'Tungro': 'టుంగ్రో తెగులు',
    'Brown Spot': 'గోధుమ రంగు మచ్చ తెగులు'
};

class SummaryEngine {
    generate(data) {
        const enShort = translations.en.short_template(data.disease, data.risk_level);
        const enLong = translations.en.long_template(data);

        // Prepare te data
        const teData = {
            ...data,
            risk_level: levelMapTe[data.risk_level] || data.risk_level,
            disease: diseaseMapTe[data.disease] || data.disease,
            reasons_te: data.reasons_te || data.reasons // fallback if no specific te reasons
        };

        const teShort = translations.te.short_template(teData.disease, teData.risk_level);
        const teLong = translations.te.long_template(teData);

        return {
            english_text_short: enShort,
            english_text_long: enLong,
            telugu_text_short: teShort,
            telugu_text_long: teLong
        };
    }
}

module.exports = new SummaryEngine();
